src/templates/CentralTemplateProvider.ts

```diff
@@ -106,6 +107,12 @@ export class CentralTemplateProvider implements Disposable {
         }
     }
 
+    public async getProjectTemplateKey(projectPath: string | undefined, language: ProjectLanguage, version: FuncVersion, projectTemplateKey: string | undefined): Promise<string> {
+        const cachedProviders = await this.getCachedProviders(projectPath, language, version, projectTemplateKey);
+        const provider = nonNullValue(cachedProviders.providers[0], 'firstProvider');
```

Comment:
Why do we only care about the firstProvider?

Reply:
Add a comment in the code. .NET only has one provider and I'm not sure how we'll handle this once we get to other languages using project keys